flink-cdc-cli/src/test/resources/definitions/pipeline-definition-full.yaml

```diff
@@ -53,3 +53,8 @@ pipeline:
   name: source-database-sync-pipe
   parallelism: 4
   enable-schema-evolution: false
+  execution.checkpointing.interval: 10000
+  execution.checkpointing.mode: EXACTLY_ONCE
+  # yarn config
+  yarn.staging-directory: /tmp/flink-cdc
+  yarn.application.queue: flink-cdc
```

Comment:
Flink 1.19 has a significant change to the format of configuration (see [FLIP-366](https://cwiki.apache.org/confluence/display/FLINK/FLIP-366%3A+Support+standard+YAML+for+FLINK+configuration)), so actually I have some concerns of this functionality now. If we still use the legacy flattened format, there will be a gap between us and the Flink. 

Another thing in my mind is that how to resolve conflicts between pipeline configs and Flink configs. There are two level of conflict:

1. Naming conflict: what if a pipeline config has the same key as Flink but with totally different meaning
2. Semantic conflict: `parallelism` in pipeline config v.s. `parallelism.default` in Flink config. Which one has higher priority and which one to choose for users?

If we mix them together in the `pipeline` section, I'm afraid there might be a lot of explanations and conflict resolving works to do in the future.

Reply:
@PatrickRen Thanks for your comment. Actually, we can use standard YAML for Flink configuration by adding a YAML config translator. As for the conflict between pipeline configs and Flink configs, how about add a new ref, like 'env-config', for flink configs, which applying for flink execution env.

Reply:
I'm afraid this will make the YAML even more complex, with so many nested layers. And the name `env-config` is quite vague. Currently we read all Flink configurations from `FLINK_HOME/conf/flink-conf.yaml(config.yaml)`. What about letting users set Flink configurations directly there?

Reply:
I don't think this is a good idea. We might need to configure different Flink configurations for different tasks, and provide users with an entry point to adjust Flink configuration based on their tasks, such as checkpointing or task manager memory. If we use the same 'flink-conf.yaml' for every task, it will become very inflexible.